packages/swingset-liveslots/src/virtualObjectManager.js

```diff
@@ -978,14 +991,20 @@ export const makeVirtualObjectManager = (
     // and into method-invocation time (which is not).
 
     let proto;
+    /** @type {ClassContextProvider | undefined} */
+    let contextProviderVar;
+    /** @type { Record<string, KitContextProvider> | undefined } */
+    let contextProviderKitVar;
+
     if (multifaceted) {
-      /** @type { Record<string, KitContextProvider> } */
-      const contextProviderKit = fromEntries(
+      contextProviderKitVar = fromEntries(
         facetNames.map((name, index) => [
           name,
           rep => {
             const vref = getSlotForVal(rep);
-            assert(vref !== undefined);
+            if (vref === undefined) {
+              return undefined;
+            }
```

Comment:
@FUDCo , this is a semantic change to existing virtual kind functionality. Is it still correct?

Reply:
I checked, and only my own code consumes this, which I know is prepared for these `undefined`s. So it cannot be an incorrect change based on code I don't know. Resolving.